api/src/main/java/org/openmrs/ConceptReferenceRange.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.hibernate.envers.Audited;
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+/**
+ * A concept reference range is typically a range of a concept for certain factor(s) e.g. age, gender e.t.c. 
+ * 
+ * <p>
+ * The criteria is used to evaluate if certain attributes of a patient meet a certain factor range. 
+ * For example, if criteria is of factor age(say age between 1-5), then the ranges only apply for this age group. 
+ * </p>
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "concept_reference_range")
+public class ConceptReferenceRange extends BaseReferenceRange {
```

Comment:
This needs to implement the standard `OpenmrsMetadata` interface I think.

Reply:
It would also be useful to be able to go from a specific `conceptNumeric` to the reference range.

Reply:
@ibacher  do you suggest that we should add a 1:1 relationship with `ConceptNumeric`?

Reply:
Oooh... good point. I guess it'd be a 1:many wouldn't it? Maybe we don't need to add that in.

Reply:
@ibacher i like your suggestion. It makes lots of sense to me. 👍

Reply:
I bumped into a blocker on this. `ConceptNumeric` have 'concept_id' as the primary key (it also extends concept which is okay). However, I already have a 1:1 relationship with `Concept`. Should I have these 2 relationships in `ConceptReferenceRange`?

Reply:
cc @dkayiwa

Reply:
@dicksonmulli what i understood this as was that, a `ConceptNumeric` gets a new property which is a list of reference ranges.

Reply:
Just the 1:1 relationship with ConceptNumeric. A ConceptNumeric is a Concept with a numeric value, which are the only concept values we have reference ranges for.